### PR TITLE
#[166240839] Users should be able to share articles across different channels

### PR DESCRIPTION
**Pull request by MCFrank16**
#### Usecase
[usecase link](https://docs.google.com/document/d/17DU9wCaLMOwjY6ynhosuXFu23xrPr0ZAxVamyBcbGCE/edit)
#### What does this PR do?

this PR opens up a way of sharing the article across Authors-Haven using the following social media platform.
``Facebook`` ``Twitter`` ``Linkedin`` and ``Gmail``

#### Description of Task to be completed?

Have the following endpoint working: 
POST ``/api/articles/: slug/share/where-to-share``: this endpoint will let the user be able to share an article on social media

#### How should this be manually tested?
* Clone the repo and saved it locally on your machine.
* Open the app using any Text Editor
* Run ``npm install``   to install the dependencies.
* Set up the   ``.env`` as instructed by the   ``.env-example``
* Run  ``npm  run dev``.
* Open ``Postman`` or install ``Postman``.
* check the shared article  in the postman using the following endpoint 
 ``POST`` ``/api/articles/:slug/share/where-to-share``.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166240839](https://www.pivotaltracker.com/story/show/166240839)

### CODE REVIEW

File: src/api/routes/articleRouter.js [hunk 1]
```diff
@@ -48,4 +50,28 @@ articleRouter.delete('/:slug',
   checkArticleOwner.checkOwner,
   articleController.delete);
 
+articleRouter.get('/:slug/share/facebook',
+  checkValidToken,
+  checkArticle.getArticle,
```
Comment (alainburindi) [inline src/api/routes/articleRouter.js]: what is the user getting?? I think the method is POST

File: tests/socialShare.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const slug = 'What-is-a-Version-1-UUID';
+const title = 'What is a Version 1 UUID';
+
+describe(' testing the social share endpoints', () => {
+  it('should login a user and get a token to use against the social share endpoint', (done) => {
+    chai
+      .request(app)
+      .post('/api/users/login')
+      .send({
+        email: 'alain1@gmail.com',
```
Comment (alainburindi) [inline tests/socialShare.test.js]: use the signup helper to get the token

**Review by alainburindi — CHANGES_REQUESTED** [review]

File: src/api/routes/articleRouter.js (hunk 1 above)
Comment (MCFrank16) [reply src/api/routes/articleRouter.js]: yeah you are right

File: tests/socialShare.test.js (hunk 2 above)
Comment (MCFrank16) [reply tests/socialShare.test.js]: yeah I did it

**Review by alainburindi — APPROVED** [review]

**Pull request closed without merging**